Analyses/src/StoppedParticlesFinder_module.cc

```diff
@@ -170,6 +180,14 @@ namespace mu2e {
       }
       std::cout<<"PhysicalVolumeInfoMultiCollection dump end"<<std::endl;
     }
+
+    if(!simStageThresholdConfigured_) {
+      if(vols_->empty()) {
+        throw cet::exception("BADINPUT")<<"StoppedParticlesFinder: something is wrong,"
+          " got empty PhysicalVolumeInfoMultiCollection "<<physVolInfoInput_<<std::endl;
+      }
+      simStageThreshold_ = vols_->size() - 1;  // the current simStage points to the last entry in vols_
+    }
   }
```

Comment:
If I've understood, you are using the PhysicalVolumeInfo map size to keep track of simulation stage high water mark, relying on it being augmented each stage.  This ties together 2 unrelated simulation features in a way that's not transparent, and might lead to unexpected behavior.  For instance, In earlier productions we were forced to remove PhysicalVolumeInfo maps from concatenation output as they bloated the file size, in this design those files could not be resampled if the threshold test is enabled.
Have you considered making an explicit subrun data product of simulation stage high water mark?  That would factorize the use cases.

Reply:
Thanks for pointing out that dropping the vector content (but keeping the vector) can address output bloat without breaking this design.  This works for me.